src/app/core/entity/entity-actions/entity-actions.service.ts

```diff
@@ -168,6 +175,58 @@ export class EntityActionsService {
     return true;
   }
 
+  /**
```

Comment:
let's move this whole block to the EntityEditService. You can inject EntityActionsService there to call the showSnackbarConfirmationWithUndo() if needed.